src/pages/api/multipleProduct.ts

```diff
@@ -0,0 +1,197 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import Cookies from 'cookies';
+import { ErrorSummary } from '../../components/ErrorSummary';
+import { MULTIPLE_PRODUCT_COOKIE, NUMBER_OF_PRODUCTS_COOKIE } from '../../constants/index';
+import { isSessionValid } from './service/validator';
+import { redirectToError, setCookieOnResponseObject, getDomain, redirectTo } from './apiUtils';
+import { trimInputOfWhiteSpace, isCurrency } from './periodProduct';
+
+export interface MultiProduct {
+    productName: string;
+    productNameId: string;
+    productNameError?: string;
+    productPrice: string;
+    productPriceId: string;
+    productPriceError?: string;
+    productDuration: string;
+    productDurationId: string;
+    productDurationError?: string;
+}
+
+export const getErrorsForCookie = (validationResult: MultiProduct[]): ErrorSummary => {
+    const errorsForCookie: ErrorSummary = {
+        errors: [],
+    };
+
+    validationResult.forEach(product => {
+        if (product.productDurationError) {
+            errorsForCookie.errors.push({
+                errorMessage: product.productDurationError,
+                id: product.productDurationId,
+            });
+        }
+        if (product.productNameError) {
+            errorsForCookie.errors.push({
+                errorMessage: product.productNameError,
+                id: product.productNameId,
+            });
+        }
+        if (product.productPriceError) {
+            errorsForCookie.errors.push({
+                errorMessage: product.productPriceError,
+                id: product.productPriceId,
+            });
+        }
+    });
+
+    return errorsForCookie;
+};
+
+export const containsErrors = (products: MultiProduct[]): boolean => {
+    let errorsExist = false;
+    products.forEach(product => {
+        if (product.productNameError || product.productPriceError || product.productDurationError) {
+            errorsExist = true;
+        }
+    });
+
+    return errorsExist;
+};
+
+export const checkProductDurationsAreValid = (products: MultiProduct[]): MultiProduct[] => {
```

Comment:
This would just be for the actual validation section, the loop could be left in here but we would split out the following to a shared function and have it return the error message:

```
const trimmedDuration = trimInputOfWhiteSpace(productDuration);
        let productDurationError;

        if (trimmedDuration === '') {
            productDurationError = `This field cannot be empty`;
        } else if (Number.isNaN(Number(trimmedDuration))) {
            productDurationError = `Product duration must be a whole, positive number`;
        } else if (!(Number(trimmedDuration) > 0)) {
            productDurationError = `Product duration cannot be zero or a negative number`;
        }
```